conda-recipes/llvmdev/build.sh

```diff
@@ -9,7 +9,11 @@ set -x
 LLVM_TARGETS_TO_BUILD=${LLVM_TARGETS_TO_BUILD:-"host;AArch64;AMDGPU;ARM;BPF;Hexagon;Mips;MSP430;NVPTX;PowerPC;Sparc;SystemZ;X86;XCore;RISCV"}
 
 # This is the clang compiler prefix
-DARWIN_TARGET=x86_64-apple-darwin13.4.0
+if [[ $build_platform == osx-arm64 ]]; then
```

Comment:
I don't know whether it matters in our setup, but it seems like this has the potential to do the wrong thing when cross-compiling on an M1 system for another system - it seems like the target is being set based on the build platform and not the target platform.

Note: the above is what I can tell from the PR that added this variable to conda-build: https://github.com/conda/conda-build/pull/4005. The docs for the variable at https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html#env-vars say "the native subdir of the conda executable", which I can't make sense of.

Reply:
I didn't review #813 that added this, but I wonder if this is potentially an issue on `main` too.